distributed/scheduler.py

```diff
@@ -3062,25 +3070,22 @@ def check_idle_saturated(self, ws: WorkerState, occ: float = -1.0):
 
         idle = self.idle
         saturated = self.saturated
-        if (
-            self.is_unoccupied(ws, occ, p)
-            if math.isinf(self.WORKER_SATURATION)
-            else not _worker_full(ws, self.WORKER_SATURATION)
-        ):
+        saturated.discard(ws)
+        if self.is_unoccupied(ws, occ, p):
             if ws.status == Status.running:
                 idle[ws.address] = ws
```

Comment:
1. saturated.discard was always called unless the worker was truly classified as such, see https://github.com/dask/distributed/blob/0dd00be5a54fc654fd5e89d4eba2d81441045b86/distributed/scheduler.py#L2602-L2616 so my behavior is consistent with what it was before
2. Other than dashboard visuals, saturated is only used in stealing to avoid sorting over all workers, https://github.com/fjetter/distributed/blob/a5d686572e3289e9d7ce71c063205cc35d4a06c2/distributed/stealing.py#L422-L431 and I'm not too concerned about this since stealing is a bit erratic either way